haas/api.py

```diff
@@ -258,34 +275,57 @@ def node_connect_network(node, nic, network):
 
     project = node.project
 
+    allocator = get_network_allocator()
+    if channel is None:
+        channel = allocator.get_default_channel(db)
+
     if nic.current_action:
         raise BlockedError("A networking operation is already active on the nic.")
 
     if (network.access is not None) and (network.access is not project):
         raise ProjectMismatchError("Project does not have access to given network.")
 
-    db.add(model.NetworkingAction(nic, network))
+    if _have_attachment(db, nic, model.NetworkAttachment.network == network):
```

Comment:
Could we do this inline `_have_attachment()`, since it is only used twice? This would be friendlier to those new to the project.

If we want to be a little more compact, perhaps we could `import NetworkAttachment`? We could also use [exists()](docs.sqlalchemy.org/en/rel_0_9/orm/query.html?highlight=exists#sqlalchemy.orm.query.Query.exists) which is a little shorter and could be more efficient on the server.